### PR TITLE
Add some parameter type checking for prepared statements

### PR DESCRIPTION
**Pull request by KevinMartin**
Whenever I use anything besides `Types.DECIMAL` for number type, I get `Error: Incorrect arguments to mysqld_stmt_execute`.

Related to: #348

### CODE REVIEW

File: lib/packets/execute.js [hunk 1]
```diff
@@ -82,11 +82,37 @@ Execute.prototype.toPacket = function ()
...
+        case 'boolean':
+          this.parameters[i] = +this.parameters[i];
+          packet.writeInt16(Types.TINY);
+          break;
+
+        case 'object':
+          if (this.parameters instanceof Date) {
+            packet.writeInt16(Types.DATETIME);
+          } else if (
+            Array.isArray(this.parameter[i]) ||
```
Comment (sidorares) [inline lib/packets/execute.js]: typo here

**Comment by sidorares** [comment]
Thanks @KevinMartin !
Could you add a test ( with `execute` +  number + string + array + date parameters ) ?

**Comment by KevinMartin** [comment]
Any guidance?

**Comment by sidorares** [comment]
Unfortunately mysql server version on TravisCI does not support json ( maybe this can be overriden, but right now .travis.yml uses default setup ). Maybe you add simple select with parameters, using at least numeric and string parameter types
Test examples:
https://github.com/sidorares/node-mysql2/blob/master/test/integration/connection/test-execute-newdecimal.js
https://github.com/sidorares/node-mysql2/blob/master/test/integration/connection/test-execute-null-bitmap.js
https://github.com/sidorares/node-mysql2/blob/master/test/integration/connection/test-execute-signed.js

Actually it looks like it's already covered ( with exception for JSON case ) Maybe you add your original JSON test, but wrap it with a sql check to execute only when JSON type supported?

**Pull request closed without merging**